src/IP.php

```diff
@@ -138,6 +142,88 @@ public function __construct($ip)
         throw new \InvalidArgumentException(sprintf('Unsupported argument type: "%s".', gettype($ip)));
     }
 
+    /**
+     * @param int $ipAddress
+     *
+     * @return IP
+     */
+    public static function newFromInteger(int $ipAddress): self
+    {
+        $ip = new static(null);
+        $ip->ip = self::fromInt($ipAddress);
```

Comment:
Following-up my previous comment: this isn't a good approach, we shouldn't build an invalid object. Instead you could pass the GMP object to the constructor. That would require a bit of refactoring of the constructor to not constantly re-test the type (GMP test could be moved at the very beginning for example), and also maybe moving when isValid is called, so it's only called once in the constructor.

Basically instead of having "conversion+verification" coupled together for each type, we would separate 1. conversion to GMP, and 2. test the GMP object.